### DISCUSSION TITLE
How to properly use Gunicorn with AWS Fargate?

### DISCUSSION THREAD

**Discussion by ihorm5** [benoitc/gunicorn#2983] [General]
The documentation states that:
"in AWS an EBS root instance volume may sometimes hang for half a minute and during this time Gunicorn workers may completely block in os.fchmod. os.fchmod may introduce extra delays if the disk gets full. Also, Gunicorn may refuse to start if it can’t create the files when the disk is full.
Currently to avoid these problems you can use a tmpfs mount"

Unfortunately, currently, tmpfs is not supported by **Fargate**, so I wonder what would be a good way to overcome this limitation? 
We have only noticed a big number of System Exit(1) events. We haven't profiled it yet, but as a temporary workaround, we have reduced the number of workers to 1 per instance(instead of recommended 2 * CPU + 1). It helped, but it would be cool if we were able to run this using several workers.

**GUNICORN_TIMEOUT** is 60s
**keep-alive** is not set
**Worker type** is **sync**

**Comment by benoitc** [comment]
farget allows you to start a temporary memory storage using shm I think This is the only way I know right now to use it.
